File: pset1/greedy.c
#include <cs50.h>
#include <math.h>
#include <stdio.h>

int main(void)
{
  float change, rounded;
  // get user input
  do
  {
    printf("Change required: ");
    change = get_float();
  }
  while (change < 0.00);
  
  //round user input
  rounded =  roundf(change * 100); 
  
  //counter
  int coinsRequired = 0;
  
  
    while (rounded >= 25)
    {
      rounded = rounded - 25;
      coinsRequired = coinsRequired + 1;
    }
    while (rounded >= 10)
    {
      rounded = rounded - 10;
      coinsRequired = coinsRequired + 1;
    }
    while (rounded >= 5)
    {
      rounded = rounded - 5;
      coinsRequired = coinsRequired + 1;
    }
    while (rounded > 0)
    {
      rounded = rounded - 1;
      coinsRequired = coinsRequired + 1;
    }
  printf("%i\n",coinsRequired);
}

File: pset6/greedy.py
# calculate minimum coins necessary to give someone change
# http://docs.cs50.net/problems/greedy/greedy.html

def main():
    while True:
        change = float(input('Change Required: '))
        if change > 0:
            break

    counter = 0
    while change > 0.25:
        change = change - 0.25
        counter += 1
    while change > 0.10:
        change = change - 0.10
        counter += 1
    while change > 0.05:
        change = change - 0.05
        counter += 1
    while round(change, 2) > round (0, 2):  # round needed to avoid errors due to floating point approximation
        change = change - 0.01
        counter += 1
    print(counter)


if __name__ == "__main__":
    main()
